server/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -276,6 +278,9 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public final XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
+        if (builder.getCompatibleMajorVersion() == Version.V_7_0_0.major) {
```

Comment:
If we are just looking at major version number, using the Version constant just to extract "7" seems overkill and less readable than just using the literal `7`.

Reply:
I think this will help the removal of compatible code. We have not decided on what mechanism to use in order to help finding this code. 
The plan at the moment is to put as much code v7-compatible into `module/rest-compatibility` 
Using Version.V_7_0_0.major would make sure that if the version is removed after major upgrade it will also highlight the places where v7-compatible  code has to be removed

Reply:
Also, the removal of the of the Version.V_7_0_0 constant in the code (for v9) will cause this to fail compilation. Which is likely the correct the behavior. It does indeed make major version bumps harder (one more thing that breaks), but for now I think this is the best we can do. It feels like there is something that can be done to ease that upgrade pain, but that is outside the scope of this PR.